[Benson15912] iP

Benson15912 opened the pull request:
# **Duke ChatBot at your service**
> When you are alone, Duke is there for you -Thelegend27

## Benefits of Duke
- Keeps track of your to-do list
- Motivates you through text
- Easy to use
- ~~Fast~~ *Crazy Fast* Responses

## All you need to do is:
1. Download it from [here] (https://github.com/Benson15912/ip/releases/tag/A-Jar)
2. double click it
3. Add your tasks
4. Let it manage your tasks and update you!


## Features
- [x] Managing Tasks
- [x] Managing Event
- [x] Managing Deadlines
- [x] GUI
- [ ] Reminders

Some commands include:
```
todo Wash Dishes
list
deadline Home /by Today
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Benson15912 committed:
Test commit

Benson15912 committed:
Added Level 0

Benson15912 committed:
Added Level-1

Benson15912 committed:
Added Level-2

Benson15912 committed:
Level 3

Benson15912 committed:
Level-4

Benson15912 committed:
Added Exceptions

Benson15912 committed:
Added 1 test case and tidied exception handling

Benson15912 committed:
Level-6 added delete function

Benson15912 committed:
Added test cases

Benson15912 committed:
Added Test cases

Benson15912 committed:
no message

Benson15912 committed:
Add Level-7

Benson15912 committed:
Add new classes for OOP

Benson15912 committed:
Add more OOP and new classes to encapsulate

Benson15912 committed:
Add Level-8 Date time feature

Benson15912 committed:
no message

Benson15912 committed:
no message

Benson15912 committed:
Merge branch 'add-gradle-support'

Benson15912 committed:
Fix Exit Code

Benson15912 committed:
Add JUnit Test

Benson15912 committed:
Add Storage OOP

Benson15912 committed:
Add Javadocs

Benson15912 committed:
Add JavaDocs

Benson15912 committed:
no message

Benson15912 committed:
no message

Benson15912 committed:
Modified Package names

Benson15912 committed:
Merge commit 'b89ca4c83527facab1a643762f6be3ce3d1a2a3b'

# Conflicts:
#	src/main/duke.txt
#	src/main/java/Parser/Parser.java
#	src/main/java/Tasks/Deadline.java
#	src/main/java/Tasks/Todo.java

Benson15912 committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

Benson15912 committed:
Fix line and indentations

Benson15912 committed:
Merge branch 'branch-A-CodingStandard'

Benson15912 committed:
no message

Benson15912 committed:
Merge branch 'master' of https://github.com/Benson15912/ip

* 'master' of https://github.com/Benson15912/ip:
  Fix line and indentations
  Modified Package names
  no message
  no message
  Add JavaDocs
  Add Javadocs
  Add Storage OOP
  Add JUnit Test

# Conflicts:
#	src/main/duke.txt

Benson15912 committed:
Add savefile creation

Benson15912 committed:
Add Level-9

Benson15912 committed:
Reformat code according to checkstyle

Benson15912 committed:
no message

Benson15912 committed:
Merge branch 'master' of https://github.com/Benson15912/ip

* 'master' of https://github.com/Benson15912/ip:
  Reformat code according to checkstyle
  Add Level-9
  Add savefile creation

# Conflicts:
#	.gitignore
#	build.gradle
#	src/main/duke.txt

vnnamng commented on src/main/java/actions/CreateDeadline.java:
```diff
@@ -0,0 +1,44 @@
+    @Override
+    public void execute(Duke bot) {
+        Deadline d = new Deadline(desc, deadline);
+        bot.getTaskList().addToList(d);
```
should this class know the bot have a tasklist?

vnnamng commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,55 @@
+     *
+     * @return The command format for saving the event task to a file.
+     */
+    public String getCommand() {
+        return String.format("event %s /from %s /to %s\n%b\n", this.description, this.from, this.to, this.isDone);
+    }
```
i think you should rename it for better clarification.

vnnamng commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+     *
+     * @return The command format for saving the deadline task to a file.
+     */
+    public String getCommand() {
+        return String.format("deadline %s /by %s\n%b\n", this.description, this.deadline, this.isDone);
+    }
```
Same as below

vnnamng commented on src/main/java/actions/Find.java:
```diff
@@ -0,0 +1,38 @@
+            for (Task task : matches) {
+                System.out.println(String.format("%d. [%s] [%s] %s", index, task.getTypeIcon(), task.getStatusIcon(),
+                        task.getDescription()));
+                index++;
```
index should have a better naming

vnnamng commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,128 @@
+            if (splitCommand.length <= 1) {
+                throw new DukeException("Please write a description and a deadline for your task!");
+            }
+            String[] infoSplit = splitCommand[1].split("/by ", 2);
```
need better name

vnnamng commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,83 @@
+            while (reader.hasNextLine()) {
+                String command = reader.nextLine();
+                String isDone = reader.nextLine();
+                Action a = parseCommand(command);
```
a need better name

vnnamng reviewed (commented):
LGTM

Benson15912 committed:
Add Level-10 and GUI support

Benson15912 committed:
Update gitignore

Benson15912 committed:
Parser Class: Reduce lines of code in method

The method parseCommand is way too long and violates coding standard.

Extracted logic part in parseCommand and placed it in their respective action classes.

By putting the logic part in the action class, the method can be shorter and more succinct

Benson15912 committed:
Add assertions to actions classes

Benson15912 committed:
Merge pull request #1 from Benson15912/branch-A-Assertions

Add assertions to actions classes

Benson15912 committed:
Add function to detect duplicates and prevent them

Benson15912 committed:
Add User Guide and reformatted Try Catch Blocks

Benson15912 committed:
Update README.md

Benson15912 committed:
Update README.md
